src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -152,6 +152,10 @@ private BoundExpression CompareWithConstant(BoundExpression input, BoundExpressi
                     return _factory.ObjectEqual(_factory.Convert(systemObject, input), boundConstant);
                 }
             }
+            else if (input.Type.IsNullableType() && boundConstant is BoundDefaultExpression boundDefault && boundDefault.NullableNeverHasValue())
```

Comment:
@gafter Ah, didn't see that `NullableNeverHasValue` was on `BoundExpression`.  Neat.  Fixed.